apps/api-journeys/src/app/modules/event/step/step.resolver.ts

```
@@ -9,21 +9,40 @@ import {
   StepViewEventCreateInput
 } from '../../../__generated__/graphql'
 import { EventService } from '../event.service'
+import { BlockService } from '../../block/block.service'
+import { VisitorService } from '../../visitor/visitor.service'
 
 @Resolver('StepViewEvent')
 export class StepViewEventResolver {
-  constructor(private readonly eventService: EventService) {}
+  constructor(
+    private readonly eventService: EventService,
+    private readonly blockService: BlockService,
+    private readonly visitorService: VisitorService
+  ) {}
+
   @Mutation()
   @UseGuards(GqlAuthGuard)
   async stepViewEventCreate(
     @CurrentUserId() userId: string,
     @Args('input') input: StepViewEventCreateInput
   ): Promise<StepViewEvent> {
+    const block: { journeyId: string } = await this.blockService.get(
+      input.blockId
+    )
+    const journeyId = block.journeyId
```

need to throw a UserInput Error here if no block.